lib/realtime/server.ex

```diff
@@ -379,14 +384,20 @@ defmodule Realtime.Server do
       _ = :ets.insert(ets, {{:route_id, route_id}, active_vehicles_and_ghosts})
     end
 
-    all_vehicles = Enum.uniq(all_vehicles(vehicles_by_route_id) ++ shuttles)
+    {logged_out_vehicles, logged_in_vehicles} =
```

Comment:
I'm considering moving this partitioning up a level to `handle_cast({:update_vehicles...`, to ensure that the `active...` and `new_active...` values don't include any data for logged out vehicles. They shouldn't because `route_id`, `run_id`, and `block_id` should all be null for logged out vehicles, but I think it may be helpful to it more explicit.

Reply:
Yeah, I like that idea! I wonder if you could go a bit further and move it to `Concentrate.Consumer.VehiclePositions.handle_events` so that the logged out vehicles are passed in as a separate value to `Server.update_vehicles`, thereby also getting rid of the thing where logged out vehicles get included in the map of vehicles by route but under the `nil` key.

Reply:
Oh nice yeah that would be even cleaner! I'll try that